packages/sdk/src/resolve/resolve-arguments.js

```diff
@@ -17,7 +21,8 @@ function cast(arg) {
 export async function resolveArguments(ix) {
   if (isTransaction(ix) || isScript(ix)) {
     for (let [id, arg] of Object.entries(ix.arguments)) {
-      ix.arguments[id].asArgument = cast(arg)
+      const castedArg = await cast(arg)
```

Comment:
I believe the attempt to resolve the `arg` through calling `arg.resolve()` should happen before it is passed into `cast` 

Should help abstract in the case that we add additional `xform`s to arguments (unlikely but possisble)

Reply:
Agreed. Changed.